src/com/goide/GoDocumentationProvider.java

```diff
@@ -102,6 +104,64 @@ else if (type == GoParserDefinition.MULTILINE_COMMENT) {
         }
         return text;
       }
-    }), "<br/>") + "</pre>";
+    }), "<br/>");
+  }
+
+  @NotNull
+  private static String getSignature(PsiElement element) {
+    if (!(element instanceof GoSignatureOwner)) {
+      return "";
+    }
+
+    PsiElement identifier = null;
+    if (element instanceof GoNamedSignatureOwner) {
+      identifier = ((GoNamedSignatureOwner)element).getIdentifier();
+    }
+    GoSignature signature = ((GoSignatureOwner)element).getSignature();
+
+    if (identifier == null && signature == null) {
+      return "";
+    }
+
+    String result = " <b>func " + (identifier != null ? identifier.getText() : "") + "(";
+
+    if (signature != null) {
+      result += getParametersAsString(signature.getParameters());
+    }
+
+    result += ")";
+
+    if (signature != null && signature.getResult() != null && signature.getResult().getParameters() != null) {
+      String signatureParameters = getParametersAsString(signature.getResult().getParameters());
+
+      if (signatureParameters.length() > 0) {
+        result += " (" + signatureParameters + ")";
+      }
+    }
+
+    return result + "</b><br/>";
+  }
+  
+  private static String getParametersAsString(GoParameters parameters) {
```

Comment:
As an example you can use https://github.com/JetBrains/intellij-community/blob/master/java/java-impl/src/com/intellij/codeInsight/hint/api/impls/AnnotationParameterInfoHandler.java, take a look at static `updateUIText` method.